.github/workflows/pull-request.yaml

```diff
@@ -10,3 +10,62 @@ jobs:
     name: PR
     uses: canonical/observability/.github/workflows/charm-pull-request.yaml@main
     secrets: inherit
+
+  test-against-coordinator:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set CWD ${{ runner.home }}
+      - run: cd ${{ runner.home }}
+
+      - name: Checkout the canonical/tempo-worker-k8s-operator repository (this branch)
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+          path: "worker"
+          # leave other args blank: it's checking out THIS branch/THIS repo
+
+      - name: Checkout the canonical/tempo-coordinator-k8s-operator repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+          repository: canonical/tempo-coordinator-k8s-operator
+          path: "coordinator"
+          ref: main
+
+      - name: Set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: pip install tox~=4.2
+
+      - name: Setup Charmcraft's pip cache
+        uses: actions/cache@v4
+        with:
+          path: /home/runner/snap/charmcraft/common/cache/charmcraft/
+          key: charmcraft-cache-${{ github.job }}-${{ strategy.job-index }}-${{ github.run_id }}-${{ github.run_attempt }}
+          restore-keys: charmcraft-cache
+
+      - name: Get IP range
+        id: ip_range
+        run: |
+          echo "ip_range=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')/32" >> $GITHUB_OUTPUT
+
+      - name: Setup operator environment (k8s)
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          juju-channel: 3.4/stable
+          provider: microk8s
+          channel: 1.26-strict/stable
+          microk8s-group: snap_microk8s
+          microk8s-addons: "hostpath-storage dns rbac metallb:${{ steps.ip_range.outputs.ip_range }}"
+          charmcraft-channel: "2.x/stable"
+
+      - name: Build the worker charm
+        run: charmcraft pack -p ${{ runner.home }}/worker
+
+      - name: Run the coordinator charm's integration tests
+        run: cd coordinator; tox -vve integration
+        env:
+          TEMPO_WORKER_CHARM: ${{ runner.home }}/tempo-worker-k8s_ubuntu-22.04-amd64.charm
```

Comment:
This should probably be `TEMPO_COORDINATOR_CHARM`, right? if so, the `.charm` filename is also wrong

Reply:
nope, that is the path to the worker charm, that we pass to the coordinator's itests to tell them to deploy this as worker instead of fetching it from charmhub